Model/ShippingProtectionTotalRepository.php

```diff
@@ -242,4 +242,23 @@ public function saveAndResaturateExtensionAttribute (
             }
         }
     }
+
+    /**
+     * Save Shipping Protection total using Magento quote ID and cart ID as provided via API
+     *
+     * @param string $cartId
+     * @param string $spQuoteId
+     * @param float $price
+     * @param string $currency
+     * @param float|null $basePrice
+     * @param string|null $baseCurrency
+     * @return void
+     * @throws AlreadyExistsException
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    public function saveByApi(string $cartId, string $spQuoteId, float $price, string $currency, float $basePrice = null, string $baseCurrency = null): void
+    {
+        $this->save($cartId, ShippingProtectionTotalInterface::QUOTE_ENTITY_TYPE_ID, $spQuoteId, ($price / 100), $currency, ($basePrice / 100) ?: null, $baseCurrency ?: null);
```

Comment:
We're making a hard assumption related to internationalization with the `/ 100` cent conversion? Nothing necessarily to change, but might be something to note for later if/when we need to update the module?

Reply:
@aarondeane-extend what happens if $basePrice is null?  Will it error out trying to be divisible by 100?

Reply:
should we update the `saveBySdk` function as well? This was copied from that method